sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -814,4 +816,50 @@ object DDLUtils {
       }
     }
   }
+
+  /**
+   * ALTER TABLE ADD COLUMNS command does not support temporary view/table,
+   * view, or datasource table with text, orc formats or external provider.
+   */
+  def verifyAlterTableAddColumn(
+      catalog: SessionCatalog,
+      table: TableIdentifier): CatalogTable = {
+    if (catalog.isTemporaryTable(table)) {
+      throw new AnalysisException(
+        s"${table.toString} is a temporary VIEW, which does not support ALTER ADD COLUMNS.")
+    }
+
+    val catalogTable = catalog.getTableMetadata(table)
```

Comment:
Call `getTempViewOrPermanentTableMetadata` instead of `getTableMetadata`. Then, you do not need the above check for temporary views. In addition, it also covers the cases for global views.